### PR TITLE
Store block hash in poc receipts txn for poc v 10

### PR DESCRIPTION
**Pull request by Vagabond**
Depends on helium/proto#52

### CODE REVIEW

File: src/transactions/v1/blockchain_txn_poc_receipts_v1.erl [hunk 1]
```diff
@@ -1212,23 +1234,31 @@ get_channels(Txn, Chain) ->
...
+                                                              blockchain_block:hash_block(Block);
+                                                          _ ->
+                                                              undefined
+                                                      end;
+                                                 (_, _Hash) -> return
+                                              end, undefined, Head, Chain)
+    end,
+
     case BlockHash of
         undefined ->
```
Comment (evanmcc) [inline src/transactions/v1/blockchain_txn_poc_receipts_v1.erl:1263]: will this be `<<>>` or `undefined` for pocv10?

File: src/transactions/v1/blockchain_txn_poc_receipts_v1.erl [hunk 2]
```diff
@@ -15,12 +15,14 @@
...
 -export([
     new/4,
+    new/5,
     hash/1,
     onion_key_hash/1,
     challenger/1,
     secret/1,
     path/1,
     fee/1,
+    block_hash/1,
```
Comment (vihu) [inline src/transactions/v1/blockchain_txn_poc_receipts_v1.erl:25]: I think we should rename this to req_block_hash just to be extra clear

File: src/transactions/v1/blockchain_txn_poc_receipts_v1.erl [hunk 3]
```diff
@@ -314,21 +332,25 @@ check_is_valid_poc(Txn, Chain, RunValidation) ->
...
                                                                                       IntData rem 8
                                                                               end, LayerData),
-                                                            %% We are on poc v9, check whether we need to run
-                                                            %% validation (presumably invoked from is_valid)
-                                                            case RunValidation of
-                                                                false ->
-                                                                    %% no need to run validations
+                                                            %% We are on poc v9
+                                                            %% %% run validations
+                                                            Ret = case POCVer > 10 of
```
Comment (vihu) [inline src/transactions/v1/blockchain_txn_poc_receipts_v1.erl:337]:
```suggestion
                                                            Ret = case POCVer >= 10 of
```

File: src/transactions/v1/blockchain_txn_poc_receipts_v1.erl (hunk 1 above)
Comment (Vagabond) [reply src/transactions/v1/blockchain_txn_poc_receipts_v1.erl:1263]: I don't remember

**Review by evanmcc — APPROVED** [review]

**Review by vihu — APPROVED** [review]

**Pull request merged**